### PR TITLE
Update: Add @yavin/model-types with ts defs for json:api queries

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Adds a new library with TS definitions to help when querying the webservice

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/model-types/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+jsonApi.included[0].type === 'dashboardWidgets';
+jsonApi.included[0].attributes.requests[0];
+```
+
+Here are more examples of fetching resources and how to type the response
+
+| fetch('\*')                            | Response Type                              |
+| -------------------------------------- | ------------------------------------------ |
+| /reports                               | JSONAPI<Report[]>                          |
+| /report/12                             | JSONAPI\<Report>                           |
```
Comment (jkusa) [inline packages/model-types/README.md:29]:
```suggestion
| /report/12                             | JSONAPI<Report>                           |
```
??

Comment (kevinhinterlong) [reply packages/model-types/README.md:29]: I'm not sure why it needs to be escaped, but it doesn't render correctly without it

Edit this comment and check out the source

| fetch('\*')    | Response Type       |
| ------------ | -------------------|
| /reports/12  | JSONAPI<Report>  |
| /report/12    | JSONAPI\<Report> |

File: packages/model-types/lib/models/delivery-rule.d.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+type BaseFormatTypes = 'csv' | 'pdf' | 'png';
+
+type GsheetFormat = {
+  type: 'gsheet';
+  options: {
+    overwriteFile: boolean;
+  };
+};
+
+type Format = GsheetFormat | BaseFormat<BaseFormatTypes>;
```
Comment (jkusa) [inline packages/model-types/lib/models/delivery-rule.d.ts:21]: Why do we have base formats and gsheet?

**Review by jkusa — APPROVED** [review]
Just the one question, otherwise 👍

**Review by jkusa — APPROVED** [review]

**Pull request merged**